components/ac_log/src/settable_log.rs

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use lazy_static::lazy_static;
+use std::sync::{Once, RwLock, ONCE_INIT};
+
+use log::Log;
+
+struct SettableLog {
```

Comment:
I feel like this is one layer too many of abstraction.  That is, you're wrapping an arbitrary `Log` implementation -- but I don't think anybody is going to want to set two _different_ implementations.  We might want to "turn off" the existing one... but we can do that by making `callback: Option<Callback>` or whatever.

Do you really think this is valuable?

Reply:
> We might want to "turn off" the existing one... but we can do that by making callback: Option<Callback>

Not being able to re-set the logger after disabling it makes writing tests for everything impossible. E.g. we can't test both `disable` and "returning false/throwing disables" (and without the tests, I had a bug in the 2nd).

It's actually much more complex than `Option<Callback>` if we want to release the native thread when the logging is disabled.

I separated it like this to make sure it's clear that it can be understood in isolation (I don't want to make the other parts of the code more complex).

We could hardcode more of this to work on `ac_log::LogSink` instead of `Box<dyn Log>` but that seems like a pretty minor win.